com.unity.render-pipelines.core/Runtime/XR/XRSystem.cs

```diff
@@ -0,0 +1,372 @@
+        /// </summary>
+        static public bool singlePassAllowed { get; set; } = true;
+
+        /// <summary>
+        /// If true, the system will log some information about the layout to the console.
+        /// </summary>
+        static public bool dumpDebugInfo { get; set; } = false;
+
+        /// <summary>
+        /// Use this method to assign the shaders that will be used to render occlusion mesh for each XRPass and the final mirror view.
+        /// </summary>
+        /// <param name="occlusionMeshPS"></param>
+        /// <param name="mirrorViewPS"></param>
+        public static void Initialize(Shader occlusionMeshPS, Shader mirrorViewPS)
+        {
+            RefreshDeviceInfo();
+
+            if (occlusionMeshPS != null)
+                s_OcclusionMeshMaterial = CoreUtils.CreateEngineMaterial(occlusionMeshPS);
+
+            if (mirrorViewPS != null)
+                s_MirrorViewMaterial = CoreUtils.CreateEngineMaterial(mirrorViewPS);
+
+            if (XRGraphicsAutomatedTests.enabled)
+                SetLayoutOverride(XRGraphicsAutomatedTests.OverrideLayout);
+        }
+
+        /// <summary>
+        /// Used by the render pipeline to communicate to the XR device how many samples are used by MSAA.
+        /// </summary>
+        /// <param name="msaaSampleCount"></param>
+        public static void SetDisplayMSAASamples(MSAASamples msaaSamples)
+        {
+            if (s_MSAASamples == msaaSamples)
+                return;
+
+            s_MSAASamples = msaaSamples;
+
+        #if ENABLE_VR && ENABLE_XR_MODULE
+            SubsystemManager.GetInstances(s_DisplayList);
+
+            foreach (var display in s_DisplayList)
+                display.SetMSAALevel((int)s_MSAASamples);
+        #endif
+        }
+
+        /// <summary>
+        /// Returns the number of samples (MSAA) currently configured on the XR device.
+        /// </summary>
+        /// <returns></returns>
+        public static MSAASamples GetDisplayMSAASamples()
+        {
+            return s_MSAASamples;
+        }
+
+        /// <summary>
+        /// Used by the render pipeline to scale the render target on the XR device.
+        /// </summary>
+        /// <param name="renderScale"></param>
+        public static void SetRenderScale(float renderScale)
+        {
+        #if ENABLE_VR && ENABLE_XR_MODULE
+            SubsystemManager.GetInstances(s_DisplayList);
+
+            foreach (var display in s_DisplayList)
+                display.scaleOfAllRenderTargets = renderScale;
+        #endif
+        }
+
+        /// <summary>
+        /// Used by the render pipeline to initiate a new rendering frame through a XR layout.
+        /// </summary>
+        public static XRLayout NewLayout()
+        {
+            RefreshDeviceInfo();
+
+            if (s_Layout.GetActivePasses().Count > 0)
+            {
+                Debug.LogWarning("Render Pipeline error : XRSystem.EndLayout() was not called!");
```

Comment:
The warning is confusing. Could we be explicit that you are calling the function right after?

Reply:
will change 👍